lib/core/request.js

```diff
@@ -97,7 +99,7 @@ class Request {
 
     this.upgrade = upgrade || null
 
-    this.path = path
+    this.path = params && Object.keys(params).length > 0 ? `${path}?${querystring.stringify(params)}` : path
```

Comment:
Addressed in the same way how Axios does it.

One thing to be aware of: URLSearchParams stringifies array into a single string `ELEMENT_1,ELEMENT_2`, which is non-reversible and is not how majority of users send arrays (typical pattern is x=ELEMENT_1,&x=ELEMENT_2). Should I keep the behaviour as implemented by `URLSearchParams` and leave it to users to prestringify their data the way they need, should I add optional array stringifier param, or implement more conventional array conversion within undici itself?

Reply:
> URLSearchParams stringifies array into a single string ELEMENT_1,ELEMENT_2

That's just because URLSearchParams doesn't care about the value type. I guess it does something equivalent to `String(value)` and that's arrays are stringified. I don't think we necessarily need to do the same thing as URLSearchParams

Reply:
@targos What would be your preference? `x=ELEMENT_1&x=ELEMENT_2` conversion for arrays?

Reply:
yes

Reply:
Done.